### PR TITLE
New Sample: Spring Boot 3 + Ehcache 3 + AOP + Metrics + Prometheus + Actuator + Vaadin

### PR DESCRIPTION
**Pull request by mathieucarbou**
## Overview

This sample integrates together:

* Spring Boot 3
* Ehcache 3
* AOP
* Metrics
* Prometheus
* Actuator
* Vaadin

It accesses an online service for currency conversion and simulate some additional latencies.

It also creates 2 caches (heap only), with listeners and TTL of 60 seconds.

## How to run this sample:

1. Requires: **Java 17**
2. You must first get an API key at https://freecurrencyapi.com/
2. Build: `./mvnw package`
3. Run: `./target/spring-boot-ehcache-0.0.1-SNAPSHOT.jar --app.freecurrencyapi.key=<your-api-key>`
4. Open your browser at http://localhost:8080/
5. Execute some currency conversion
6. Look at the logs below
7. Look at Spring Boot actuator endpoints at http://localhost:8080/actuator to see the metrics and caches and verify
   that the caching works.

### CODE REVIEW

**Review by jhouserizer — APPROVED** [review]
looks good!

**Review by GaryWKeim — APPROVED** [review]

**Comment by GaryWKeim** [comment]
Just one more thing (Columbo reference)... are we sure that the people requesting this demo will want to use Java 17 or maybe something older?

File: 08-spring-boot-ehcache/src/main/java/com/example/springbootehcache/MainView.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+@Route
+@PreserveOnRefresh
+@SpringComponent
+public class MainView extends VerticalLayout {
+  @Serial
+  private static final long serialVersionUID = 6004653986145050428L;
+
+  @Autowired
+  public ExchangeService exchangeService;
```
Comment (chrisdennis) [inline 08-spring-boot-ehcache/src/main/java/com/example/springbootehcache/MainView.java:40]: Should we using constructor rather than field injection?

**Review by chrisdennis — COMMENTED** [review]
If we're going to go Java 17... should we using `var` and avoiding some of the more redundant and excessively long type names?

File: 08-spring-boot-ehcache/src/main/java/com/example/springbootehcache/MainView.java (hunk 1 above)
Comment (mathieucarbou) [reply 08-spring-boot-ehcache/src/main/java/com/example/springbootehcache/MainView.java:40]: That's not the only one.
My take on that: I prefer constructor injection, but for a PoC/Sample like that, I prefer readability / less code.
Just my opinion.

**Comment by mathieucarbou** [comment]
> names

I can make a second pass tomorrow yes to better use java 17 features. I would like it too!

**Pull request merged**